owner-onboarding-server/src/ov_management/mod.rs

```diff
@@ -0,0 +1,242 @@
+use std::{sync::Arc, collections::HashMap};
+use openssl::pkey::{Private, PKey};
+use warp::{Filter};
+use std::{usize,str::FromStr};
+use fdo_data_formats::{ownershipvoucher::OwnershipVoucher, types::Guid, DeserializableMany};
+use serde::{Serialize, Deserialize};
+use warp::{hyper::StatusCode, http::HeaderValue, Buf, reply::Json};
+use crate::OwnerServiceUD;
+
+pub(crate) fn ov_filter(session_store:Arc<OwnerServiceUD>) 
+-> impl Filter<Extract = impl warp::Reply, Error = warp::Rejection> + Clone
+{       
+        let store1 = session_store.clone();
+        let baseurl = warp::path("management").and(warp::path("v1"));
+        let add = baseurl        
+        .and(warp::post())
+        .and(warp::path("ownership_voucher"))
+        .and(warp::path::end())
+        .and(warp::header::value("X-Number-Of-Vouchers"))
+        .and(warp::header::value("Content-Type"))
+        .and(warp::body::content_length_limit(1024 * 32))
+        .and(warp::body::bytes())
+        .map(move |count,content_type, body| 
+                (count,content_type,body,session_store.clone()))
+        .untuple_one()
+        .and_then(add_ov);
+
+        let delete = baseurl
+        .and(warp::post())
+        .and(warp::path("ownership_voucher"))
+        .and(warp::path("delete"))
+        .and(warp::path::end())
+        .and(warp::header::exact("Content-Type", "Application/json"))
+        .and(warp::body::content_length_limit(1024 * 32))
+        .and(warp::body::aggregate())
+        .map(move |uuids| (uuids,store1.clone()))
+        .untuple_one()
+        .and_then(delete_ov);
+        add.or(delete)
+
+}
+
+async fn add_ov(
+    count: HeaderValue,
+    content_type: HeaderValue, 
+    body: impl Buf,
+    user_data: Arc<OwnerServiceUD>,
+) -> Result<impl warp::Reply,warp::Rejection> 
+{   
+    // check for ov type
+    let content_type = content_type.to_str().unwrap();
+    let ovs= match content_type {
+        "application/x-pem-file" => OwnershipVoucher::many_from_pem(body.chunk()),
+        "application/cbor" => OwnershipVoucher::deserialize_many_from_reader(body.chunk()),
+        _ => {
+            let  e =  ErrorMsg::generate(ErrorCode::RequireContentTypeHeader, ACCEPTED_CONTENT_TYPE);
+            return Ok(warp::reply::with_status(e, StatusCode::BAD_REQUEST));
+        }
+    };
+
+    // parse-error
+    if ovs.is_err() {
+        let  e =  ErrorMsg::generate(ErrorCode::ParseError, ovs.unwrap_err().to_string());
+        return Ok(warp::reply::with_status(e, StatusCode::BAD_REQUEST));
+    }
+
+    // check-empty
+    let ovs = ovs.unwrap();
+    if ovs.is_empty() {
+        let  e =  ErrorMsg::generate(ErrorCode::ParseError, "Ownership voucher not found!!");
+        return Ok(warp::reply::with_status(e, StatusCode::BAD_REQUEST));
+    }
+
+    // check if X-Number-Of-Vouchers present
+    let count = count.to_str().unwrap().parse::<usize>();
```

Comment:
We shouldn't use `.unwrap()`.
Maybe do something like:
`let count = count.to_str().map_err(|_| ErrorMsg::generate(...)?.parse::<usize>().map_err(|_| ...)?;`

Reply:
I get that unwrap() will cause error incase of any error, but map_err(|_| warp::reject()) is returning ghost error with status code 500, so handled all the errors before unwrapping , which works as expected now.

Reply:
created a a new function to handle this